src/main/java/seedu/weme/statistics/TagWithCount.java

```diff
@@ -0,0 +1,46 @@
+package seedu.weme.statistics;
+
+import seedu.weme.model.tag.Tag;
+
+/**
+ * A wrapper class of {@code Tag} that carries the number of occurrence of such a tag
+ * in a {@code MemeBook}for {@code TagManager} in statistics.
+ */
+public class TagWithCount implements Comparable<TagWithCount> {
+    private Tag tag;
+    private int count;
+
+    /**
+     * Constructs a {@code TagWithCount}
+     * @param tag
+     * @param count
+     */
+    public TagWithCount(Tag tag, int count) {
+        this.tag = tag;
+        this.count = count;
+    }
+
+    public Tag getTag() {
+        return tag;
+    }
+
+    public int getCount() {
+        return count;
+    }
+
+    @Override
+    public String toString() {
+        return tag.tagName + " count: " + count;
```

Comment:
Maybe use StringBuilder here.

Reply:
My take is that for short string concatenation, "+" operator is good enough (Performance not affected by much). Would create a problem if the number of tags in stats grow bigger (In that case I can't even display so i have to list the top 10 only)